department-of-veterans-affairs/caseflow
# Ki/APPEALS-42516 - Create Details Page Shell

Resolves [APPEALS-42516 - Create Details Page Shell](https://jira.devops.va.gov/browse/APPEALS-42516)

# Description
This is a new page that allows the pathway ".../queue/correspondence/:correspondence_id" to reach a new page for details about a correspondence matching a `correspondence_id`.
- New CorrespondenceDetails page
- New correspondence_details_controller
- Updated routes

## Acceptance Criteria

## Testing Plan
1. Navigate to correspondence queue (Your Correspondences or Correspondence Cases page)
2. Select a correspondence and notice link (i.e. http://localhost:3000/queue/correspondence/bde5e263-d85b-4e8a-a49d-4a158087fc82/review_package)
3. Remove the review_package (or any other after-pathways) to navigate to the correspondence details page
    1. i.e. **_http://localhost:3000/queue/correspondence/bde5e263-d85b-4e8a-a49d-4a158087fc82_**
4. Check that details page has relevant details:
    1. Veteran Details
    2. Veteran ID (FileNumber)
        1. This is a "clickable copy box".  Validate that clicking on it properly copies the number into your clipboard and can be pasted somewhere else.
    3. View All Correspondence Link (currently links to under_construction)
    4. Record status: Pending (currently does not change)
    5. Has 4 Clickable Tabs
        1. Correspondence and Appeal Tasks
        2. Package Details
        3. Response Letters
        4. Associated Prior Mail
 5. Repeat test steps as necessary for different correspondences.

# Frontend
## User Facing Changes

![Screenshot 2024-07-12 at 11 22 31 AM](https://github.com/user-attachments/assets/30808b4d-9721-4b16-b742-fbb3381a04b0)

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?
